client/src/views/Models/components/DrilldownPanel/MetadataPane.vue

```diff
@@ -78,27 +115,65 @@
 <script lang="ts">
   import Vue from 'vue';
   import Component from 'vue-class-component';
-  import { Prop } from 'vue-property-decorator';
+  import { Prop, Watch } from 'vue-property-decorator';
   import _ from 'lodash';
 
+  import { emmaaEntityInfo, emmaaEvidence } from '@/services/EmmaaFetchService';
+
+  import { truncateString } from '@/utils/StringUtil';
+
   import * as GroMET from '@/types/typesGroMEt';
+  import * as EMMAA from '@/types/typesEmmaa';
+
   import { formatFullDateTime } from '@/utils/DateTimeUtil';
   import MessageDisplay from '@/components/widgets/MessageDisplay.vue';
+  import LoadingAlert from '@/components/widgets/LoadingAlert.vue';
 
   const METADATA_TYPES_ORDER = [
     GroMET.MetadataType.TextDefinition,
     GroMET.MetadataType.TextParameter,
     GroMET.MetadataType.EquationDefinition,
     GroMET.MetadataType.CodeSpanReference,
+    GroMET.MetadataType.CodeSpanReference,
+    GroMET.MetadataType.IndraAgentReferenceSet,
+    GroMET.MetadataType.ReactionReference,
   ];
 
   const components = {
     MessageDisplay,
+    LoadingAlert,
   };
 
   @Component({ components })
   export default class MetadataPane extends Vue {
     @Prop({ default: [] }) metadata: GroMET.Metadata[];
+    @Prop({ default: '' }) modelName: string;
+
+    isLoading = false;
+    agentsReferences: EMMAA.EmmaaEntityInfoInterface[] = [];
+    nodeEvidences: EMMAA.EmmaaEvidenceEvidenceInterface[] = [];
+
+    @Watch('metadata') onDataChange (): void {
+      if (this.metadata.find(datum => datum.metadata_type === GroMET.MetadataType.IndraAgentReferenceSet)) {
+        this.fetchAgentsReferences();
+      }
+      if (this.metadata.find(datum => datum.metadata_type === GroMET.MetadataType.ReactionReference)) {
+        this.fetchNodeEvidences();
+      }
+    }
+
+    mounted (): void {
+      if (this.metadata.find(datum => datum.metadata_type === GroMET.MetadataType.IndraAgentReferenceSet)) {
+        this.fetchAgentsReferences();
+      }
+      if (this.metadata.find(datum => datum.metadata_type === GroMET.MetadataType.ReactionReference)) {
+        this.fetchNodeEvidences();
+      }
+    }
+
+    get hasNodeEvidences (): boolean {
+      return this.nodeEvidences.length > 0;
```

Comment:
I got the following error while testing:
![Screen Shot 2021-08-04 at 20 04 04](https://user-images.githubusercontent.com/636801/128270885-92173471-68ee-4634-bb1d-664567579c8c.png)

Maybe on line 251 you can make sure `this.nodeEvidences` is an array by adding conditional test `?? [];`

Reply:
Done.